### DISCUSSION TITLE
Deployment with MDM

### DISCUSSION THREAD

**Discussion by steven-vn** [Homebrew/discussions#2562] [Getting started]
Hello,

I'm currently in the process of deploying an MDM (Mosyle) and I've tried running the bash command found on the Homebrew homepage which did not install homebrew on the deployed Macs. I've tried using the script, which is found at the link in the bash command (modified it so that it would use the root user to run the script as Mosyle uses the root user to deploy custom commands and scripts) and it still does not deploy Homebrew. What I'm looking to find out is how I can deploy Homebrew via script using an MDM, any help would be appreciated. 

in before: You have to use the root user to deploy anything under the MDM.

Thanks

**Comment by SMillerDev** [comment]
Just follow the instructions in the mosyle UI to run the script as user.

**Reply by steven-vn** [reply]
Tried that, did not work, besides, we like to install it system-wide and not just for the local user

**Reply by SMillerDev** [reply]
The homebrew install script does install system wide, but it is only meant to be ran by a single user.

https://docs.brew.sh/FAQ#why-does-homebrew-say-sudo-is-bad

Has a section on multi-user installs of homebrew, and as for the mosyle scripting. I think their support is better equipped to answer this.

**Comment by rfossicav** [comment]
@steven-vn Hi Steven, did you ever find a solution for this?

**Reply by bedwin42** [reply]
@rfossicav Did you figure this out by any chance?

**Reply by rfossicav** [reply]
Hi bedwin42, I haven't tested it yet (waiting for a test machine to become available for me to re-image and try this from scratch!)  but I've added [honestpuck's script](https://github.com/Honestpuck/homebrew.sh/blob/master/homebrew-3.3.sh) as a custom command in Mosyle. In theory, it should install it, eh?

**Comment by bedwin42** [comment]
Thanks sir, if you end up testing that honestpuck script, can you let me
know how it goes please?
